sklearn/tree/_criterion.pyx

```diff
@@ -676,6 +676,72 @@ cdef class Gini(ClassificationCriterion):
         impurity_right[0] = gini_right / self.n_outputs
 
 
+cdef class HellingerDistance(ClassificationCriterion):
+    r"""Hellinger distance criterion.
+
+    This handles cases where the target is a classification taking values 0 and 1.
+    If k represents a node before split, k1 and k2 represent the children nodes after split,
+    left_0 represents observations of class 0 in node k1, right_0 represents observations of class 0 in node k2,
+    left_1 represents observations of class 1 in node k1, right_1 represents observations of class 1 in node k2,
+    parent_0 represents observations of class 0 in node k, parent_1 represents observations of class 1 in node k,
+    then let
+
+        count_k1 = (\sqrt{\frac{N_{left_0}}{N_{parent_0}}}-\sqrt{\frac{N_{left_1}}{N_{parent_1}}})^2
+
+    be the partial hellinger distance score for the k1 node.
+
+        count_k2 = (\sqrt{\frac{N_{right_0}}{N_{parent_0}}}-\sqrt{\frac{N_{right_1}}{N_{parent_1}}})^2
+
+    be the partial hellinger distance score for the k2 node.
+
+    Hellinger distance score is then defined as:
+
+        hellinger_distance = \sqrt{count_k1+count_k2}
+    """
+    cdef double node_impurity(self) nogil:
+        cdef:
+            double impurity_left
+            double impurity_right
+
+        self.children_impurity(&impurity_left, &impurity_right)
+
+        return impurity_right + impurity_left
+
+    cdef void children_impurity(self, double* impurity_left,
+                                double* impurity_right) nogil:
+        cdef:
+            SIZE_t* n_classes = self.n_classes
+            double* sum_left = self.sum_left
+            double* sum_right = self.sum_right
+            double hellinger_left = 0.0
+            double hellinger_right = 0.0
+            double count_k1 = 0.0
+            double count_k2 = 0.0
+            SIZE_t k, c
+
+        for k in range(self.n_outputs):
+            count_k1 = 0.0
+            count_k2 = 0.0
+            if(sum_left[0] + sum_right[0] > 0):
```

Comment:
Each condition to check for division by zero is repeated two times.
One is used to calculate for the left child and the other for the right child.
I don't know which is better for performance: have the same condition two times but save space for the variables, or add variables for both nodes and only one condition.

Reply:
You're right, thanks
I've refactored the code to avoid the duplications